### DISCUSSION TITLE
Feature request

### DISCUSSION THREAD

**Discussion by ghost** [bulletmark/mpr#3] [General]
Hi, I find your little wrapper very useful. But the xrun function is a bit unclear for me. If I run "mpr xrun main" in my local source directory with a blank device connected it will fail because of missing subdirectories on the device.
Is there a possibility that xrun will copy the folders too?

**Comment by bulletmark** [comment]
Sorry for the slow response. I am travelling OS with my family atm.

Automatically checking for and creating the directory tree would be too slow and unreliable using `mpremote` so it is expected you first create the mirror tree yourself manually. This is trivial to do using `mpr mkdir`. After you create the tree then `mpr xrun` will automatically update/sync all `**/*.mpy` files from then on.

**Reply by ghost** [reply]
Thanks for the response. Maybe it would be nice to mension this in the readme for the xrun command. :)

**Reply by bulletmark** [reply]
Actually, at the same time as that comment from 5 days ago I also added commit 6c8772ef67e99cfa752ba965e6a971dede8d9ae4 to the README.

**Reply by ghost** [reply]
Sorry, missed this out. Thank you for clearification.
